src/goto-programs/class_hierarchy.h

```diff
@@ -57,6 +57,10 @@ class class_hierarchyt
   void operator()(const symbol_tablet &);
 
   class_hierarchyt() = default;
+  explicit class_hierarchyt(const symbol_tablet &symbol_table)
```

Comment:
:green_heart: Why even have the default constructor + operator? This is perfect!

Reply:
for cases where the user wants to allocate it in advance, e.g. as a member of another class that won't have access to the symbol table until its own `operator()`, so I won't kill the no-arg constructor. However agreed this is better when possible.